src/main/java/frc/robot/subsystems/SwerveModule.java

```diff
@@ -0,0 +1,284 @@
+  }
+
+  public void setDesiredState(SwerveModuleState desiredState, boolean isOpenLoop, boolean isDisableAntiJitter) {
+    // Custom optimize command, since default WPILib optimize assumes continuous controller which
+    // REV and CTRE are not
+
+    desiredState = SwerveModuleState.optimize(desiredState, getState().angle);
+    // desiredState.speedMetersPerSecond *= desiredState.angle.minus(getAngle()).getCos();
+
+    if (synchronizeEncoderQueued) {
+      synchronizeEncoderQueued = false;
+      resetToAbsolute();
+      // desiredState = new SwerveModuleState(0, lastAngle);
+    }
+
+    setAngle(desiredState, isDisableAntiJitter);
+    setSpeed(desiredState, isOpenLoop);
+
+
+    desiredAngleEntry.accept(desiredState.angle.getRadians());
+    desiredSpeedEntry.accept(desiredState.speedMetersPerSecond);
+    speedError.accept((desiredState.speedMetersPerSecond)-(driveEncoder.getVelocity()));
+    angleError.accept((desiredState.angle.getRadians())-(getAngle().getRadians()));
+  }
+
+  /**
+   * Resets Position Encoder
+   */
+  private void resetToAbsolute() {
+    double absolutePosition = getCanCoder().getRadians() - angleOffset.getRadians();
+    integratedAngleEncoder.setPosition(absolutePosition);
+    lastAngle = getAngle();
+  }
+
+  private void configAngleEncoder() {
+    angleEncoder.configFactoryDefault();
+    CANCoderUtil.setCANCoderBusUsage(angleEncoder, CCUsage.kMinimal);
+    angleEncoder.configAllSettings(Robot.ctreConfigs.swerveCanCoderConfig);
+  }
+
+  private void configAngleMotor() {
+    configureSpark("Angle restore factory defaults", () -> angleMotor.restoreFactoryDefaults());
+    CANSparkMaxUtil.setCANSparkMaxBusUsage(angleMotor, Usage.kPositionOnly);
+    configureSpark("Angle smart current limit", () -> angleMotor.setSmartCurrentLimit(Config.Swerve.angleContinuousCurrentLimit));
+    angleMotor.setInverted(Config.Swerve.angleInvert);
+    configureSpark("Angle idle mode", () -> angleMotor.setIdleMode(Config.Swerve.angleNeutralMode));
+    configureSpark("Angle position conversion factor", () -> integratedAngleEncoder.setPositionConversionFactor(Config.Swerve.angleConversionFactor));
+    configureSpark("Angle set P", () -> angleController.setP(Config.Swerve.angleKP));
+    configureSpark("Angle set I", () -> angleController.setI(Config.Swerve.angleKI));
+    configureSpark("Angle set D", () -> angleController.setD(Config.Swerve.angleKD));
+    configureSpark("Angle set FF", () -> angleController.setFF(Config.Swerve.angleKFF));
+    configureSpark("Angle set pid wrap min", () -> angleController.setPositionPIDWrappingMinInput(0));
+    configureSpark("Angle set pid wrap max", () -> angleController.setPositionPIDWrappingMaxInput(2 * Math.PI));
+    configureSpark("Angle set pid wrap", () -> angleController.setPositionPIDWrappingEnabled(true));
+    configureSpark("Angle enable Volatage Compensation", () -> angleMotor.enableVoltageCompensation(Config.Swerve.voltageComp));
+    resetToAbsolute();
```

Comment:
Remove this call to resetToAbsolute() on ~line 168 because it's already called in constructor of this class